### PR TITLE
315: Judgement

### PR DESCRIPTION
**Pull request by jdonkervliet**
Updated version.

Turkçe bilmiyorum. I don't speak Turkish.

### CODE REVIEW

**Comment by pimotte** [comment]
I strongly object to the use of "the player with the smallest alphanumeric GitHub handle" since puts me (and Stefan too) at quite a disadvantage.

**Comment by pimotte** [comment]
Also, multiple turns in parallel has not passed yet.

**Comment by jdonkervliet** [comment]
@pimotte see my issue(s).

**Comment by jdonkervliet** [comment]
Fixed smallest.

**Comment by MrHug** [comment]
I am okay with this rephrase, but as this PR is currently in violation of rule 201, I will abstain from voting for now.

**Comment by pimotte** [comment]
Since concurrent turns are now allowed, this is okay, however, the number of the rule is incorrect, as we just merged rule 306 as voting with thumbs.

**Comment by jdonkervliet** [comment]
Updated the number.

**Comment by MrHug** [comment]
:+1:

**Comment by pimotte** [comment]
"If the majority vote does not result in a majority vote, the vote is repeated." is phrased weirdly and could lead to infinite loops. This would need some sort of tie-breaker mechanism.

Judgement is misspelled as Judgment at various points.

"If this set contains" should probably be a new paragraph, and not be in bullet point 2.

**Comment by MrHug** [comment]
As mentioned in #23, I think this should be 307

**Comment by MrHug** [comment]
Hmm interesting question, can I retract a vote? If so, do I just do so by removing my comment? (That could be considered confusing)

EDIT: Although in this case the vote is invalid because of rule 313. (Not that it matters, because 313 states that invalid votes are still valid iff they only violate that rule.)

**Comment by pimotte** [comment]
This vote is not invalid because of 313, since it was cast before the enactment of 313 and you can't have retroactive application (107). I think. Also note that 313 says there are no sanctions for voting invalidly, not that the vote is actually valid, the vote itself is still invalid.

I'd say you can retract your vote by stating you do so by 116, as retracting a vote is an unregulated action.

**Comment by MrHug** [comment]
But isn't the vote's validity only called into question upon the tally of the votes, something that is still to happen? Also, interesting remark about the invalidity. Does that mean that merging the PR with such invalid votes is open for sanctions?

And fair enough, so I hereby _retract_ my vote.

**Comment by pimotte** [comment]
In my interpretation, yes, merging a PR with invalid votes is open for sanctions, if that merge is then without the needed majority.

With regards to votes, that is the other interpretation. If you feel strongly that the moment of tally is the moment a vote is considered to be valid/invalid, then we can take that as interpretation. In my opinion, the validity of a vote is an attribute at casting time, at least in the current ruleset, and hence cannot be affected ex post facto.

**Comment by MrHug** [comment]
Hmm well can't argue with Latin... :stuck_out_tongue_winking_eye:  But I guess I can live with your interpretation.

And I indeed misread rule 313, but I like the fact that we have a first clear case that is open for possible sanctions and thus Trials in the future :D

**Comment by jdonkervliet** [comment]
What's the status of this?

**Comment by MrHug** [comment]
The number is wrong, I think it should be 307, hence I had retracted my vote. Other than that it looked good to me.

File: README.md [hunk 1]
```diff
@@ -135,17 +135,26 @@ determining precedence.
...
+Players are in this set when all of the following constraints are met:
+ 1. The player is not the most recent editor of (one of) the rule(s) that is the
+ 		subject of the disagreement.
+ 2. The player is not currently a Judge in another Judgement.
+If this set contains multiple players, the player with the alphanumeric GitHub
+handle previous to the player who invoked Judgement is selected from this group.
+If this set is empty, the rules above are removed one-by-one in bottom-to-top order until
+the set is no longer empty.
+If all these rules have been removed, a Judge is decided by majority vote.
+If the majority vote does not result in a majority vote, the vote is repeated.
```
Comment (MrHug) [inline README.md]: Though perhaps this line should be changed ;)

A majority vote is per definition a majority vote ;)

**Comment by pimotte** [comment]
Voting is open, with one vote in favour. I have some open comments at https://github.com/pimotte/nomic/pull/12#issuecomment-219690327

Edit: Apologies. @MrHug did retract his vote. I agree there is some weirdness with the numbering too.

**Comment by MrHug** [comment]
Actually that vote in favour was retracted. As no rule for this existed at the time this was done in a comment saying:

> And fair enough, so I hereby _retract_ my vote.

**Comment by jdonkervliet** [comment]
Okay, I'll change the number and rephrase that weird sentence.

**Comment by jdonkervliet** [comment]
Done.

**Comment by MrHug** [comment]
Ah you rebased, I was already wondering about all the sudden commits ;)

File: README.md [hunk 2]
```diff
@@ -136,17 +136,26 @@ determining precedence.
...
+definition.
+
+When a player invokes Judgement, a Judge is selected to decide the question.
+The Judge is selected from the set of players that is eligible to become
+Judge.
+Players are in this set when all of the following constraints are met:
+ 1. The player is not the most recent editor of (one of) the rule(s) that is the
+ 		subject of the disagreement.
+ 2. The player is not currently a Judge in another Judgement.
+If this set contains multiple players, the player with the alphanumeric GitHub
```
Comment (MrHug) [inline README.md:151]: Extra new line before this line to make sure it is not in the bullet.

File: README.md [hunk 3]
```diff
@@ -136,17 +136,26 @@ determining precedence.
...
+handle previous to the player who invoked Judgement is selected from this group.
+If this set is empty, the rules above are removed one-by-one in bottom-to-top order until
+the set is no longer empty.
+If all these rules have been removed, a Judge is decided by majority vote.
+If the majority vote does not produce a conclusive result, the vote is repeated.
```
Comment (MrHug) [inline README.md]: Could be repeated infinitely. Perhaps something like: After `x` iterations, the player with the alphanumeric GitHub handle previous to the invoker is made Judge. (btw what if there is not previous? I think there isn't one for @arthurbik)

**Comment by jdonkervliet** [comment]
Processed feedback from @MrHug.

**Comment by jdonkervliet** [comment]
:+1:

**Comment by Jason-Statham** [comment]
More votes are needed!

**Comment by MrHug** [comment]
There has been no Japanese or Turkish yet I think, do you wanna do the honours?

**Comment by jdonkervliet** [comment]
I edited my original message. Does that count?

**Comment by MrHug** [comment]
The rule mentions a comment of the PR, but I would argue the first text of the PR is also a comment. @pimotte also used this in his own proposal (#21)

**Comment by MrHug** [comment]
:+1:

**Comment by Jason-Statham** [comment]
More votes are needed!

**Comment by pimotte** [comment]
I like involving Jason, but as far as I can see, he doesn't have this functionality yet. Do we want that before we adopt this?

**Comment by jdonkervliet** [comment]
This is trivial to make. I'm okay with just merging this now. I'll make sure to implement this in a timely fashion when this functionality is actually needed.

**Comment by pimotte** [comment]
:+1:

**Comment by Jason-Statham** [comment]
More votes are needed!

**Comment by jdonkervliet** [comment]
:fearful: oops, my points.

**Pull request merged**